### PR TITLE
[inductor] Fix a perf regression caused by https://github.com/pytorch/pytorch/pull/98214

### PR DESCRIPTION
**Pull request by desertfire**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #98343

Summary: See the description in https://github.com/pytorch/pytorch/issues/98342

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10

### CODE REVIEW

File: torch/_inductor/triton_heuristics.py [hunk 1]
```diff
@@ -138,7 +138,9 @@ def launcher({', '.join(def_args)}, grid, stream):
         launcher.n_regs = getattr(binary, "n_regs", None)
         launcher.n_spills = getattr(binary, "n_spills", None)
         launcher.shared = getattr(binary, "shared", None)
-        if config.triton.store_cubin:
+        launcher.store_cubin = config.triton.store_cubin
```
Comment (eellison) [inline torch/_inductor/triton_heuristics.py:141]: Why did this cause a regression if `config.triton.store_cubin` is false by default ?

Comment (desertfire) [reply torch/_inductor/triton_heuristics.py:141]: Here it's not causing a problem. It is at Line 236. My theory is accessing the global `config.triton.store_cubin` is expensive when we are running multi-threading (invoking kernels through `.run`).

**Review by jansel — APPROVED** [review]

File: torch/_inductor/triton_heuristics.py (hunk 1 above)
Comment (ngimel) [reply torch/_inductor/triton_heuristics.py:141]: That's very weird, it's happening during compilation only, and shouldn't affect runtime.

**Pull request closed without merging**